### PR TITLE
[Experimental] Delayed Account Creation Block

### LINKED ISSUE #50632: Delayed account creation: Create "account creation" block
Create the account creation block. This should have a feature branch until complete.

Requirements:

- Placement of the feature in Order confirmation— after initial order details
- Editable heading and bullet points
- Styles inherited from global settings
- Default settings:
     - Order email will be the account email
     - Shopper will set up a password
     - Minimum 8 character password (or adjust strengths in existing component)

As well as creating the block and making it functional on the frontend there will be additional tasks for the editor UX and default content.

This block will require API support in order to create accounts (see https://github.com/woocommerce/woocommerce/issues/48580) unless we opt to go with a basic form POST instead. TBD

Figma flows:
xsRWL4dMRYrTsMrLbXIuoA-fi-57_543

image

p2: pdFofs-1U1-p2

```[tasklist]
### Tasks
- [x] Create block
- [x] Editor settings: Background and text colors
- [x] Insert block into default order confirmation page template
- [x] Make create account button functional
- [x] Show confirmation after registration
```

### PR DESCRIPTION
**Pull request by mikejolley**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This is the delayed account creation block which will be included on the order confirmation page. This PR includes the new block and baseline functionality behind an experimental build flag.

Closes #50632

### How to test the changes in this Pull Request:

**This block is behind an experimental flag. If you are testing on a production version of WooCommerce this block should not be available (yet).** However you can test:

1. That validation errors during checkout now show a red "i" icon to the left of the text
2. That you can still place a guest email, copy paste the confirmation page URL into a different browser, then input your email address to view the order
3. That the order confirmation page displays correctly without delayed account creation visible 

These instructions are for testing the development build:

1. Appearance > Templates > WooCommerce > Order Confirmation
4. Insert the **Account Creation** block below the order summary (this is were it will be auto inserted in the future). Save.
5. Confirm the editor shows a non-interactive form and an editable list. 
6. You can change the contents of the list if you wish. 
7. Test changing background and text styles, and dark controls on and off.
8. Save changes.
9. On the store frontend, while logged out, add some items to cart and checkout as a guest. Use a new email address.
10. On the confirmation screen check that the account creation block is shown. Enter a password and submit. Ensure that:
    1. Account is created and you are now logged in
    2. Messaging links to the account area
    3. Going to the account area and viewing orders shows the order you just created
    4. That you received the new account email from WooCommerce
    5. Log-out and log-in successfully using the password/email combo you just signed up with.
11. Log-out and go through the checkout again. Checkout as a guest, but use the previous email.
12. The order confirmation should not show the account creation box because the email was already used.
13. Smoke test/try to break the create account form as best you can.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/order-confirmation/create-account/block.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		},
+		"align": {
+			"type": "string",
+			"default": "wide"
+		},
+		"className": {
+			"type": "string",
+			"default": ""
+		},
+		"hasDarkControls": {
```
Comment (alexflorisca) [inline plugins/woocommerce-blocks/assets/js/blocks/order-confirmation/create-account/block.json:27]: Out of scope for this pr, but I wonder if we can move this setting to global styles instead as it's now used in 3 different places now (Cart, Checkout and this block)

File: plugins/woocommerce/changelog/add-delayed-account-creation-block-50632 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
```
Comment (alexflorisca) [inline plugins/woocommerce/changelog/add-delayed-account-creation-block-50632:1]: I'd argue this could be a major? It's a chunky bit of new functionality

**Review by alexflorisca — CHANGES_REQUESTED** [review]
Nice work @mikejolley! Code looks good and generally tests really well. Just noticed a couple of things that stood out:

- The text for the disabled state of the button doesn't show when the button background is white
Screenshot 2024-09-09 at 15 28 12

- When I'm logged in as an admin and place an order, I still see the Create new account form.
- When the account is created, I'd expect the shipping and billing addresses from the order to be stored as my addresses, but they're not. The addresses under my account are empty.
- After creating the account and posting the form, the order confirmation page is refreshed, but the Success message is below the fold, so it's not immediately obvious what's happened. 2 ideas here. 1) Can we tag on the id of the form to the end of the post url so that it scrolls down to the form once the page is refreshed? Or 2) Can we display a notice at the top of the page instead of the account form with the success message?

**Comment by mikejolley** [comment]
Thanks @alexflorisca I think this is ready to look again.

> The text for the disabled state of the button doesn't show when the button background is white

https://github.com/woocommerce/woocommerce/pull/50934/commits/3d49ac297f122c89826f931b4340adc5e5c0bb0c

> After creating the account and posting the form, the order confirmation page is refreshed, but the Success message is below the fold, so it's not immediately obvious what's happened. 2 ideas here. 1) Can we tag on the id of the form to the end of the post url so that it scrolls down to the form once the page is refreshed? Or 2) Can we display a notice at the top of the page instead of the account form with the success message?

https://github.com/woocommerce/woocommerce/pull/50934/commits/5f35c54901873c54e9560a043f8250100a0f5523

> When the account is created, I'd expect the shipping and billing addresses from the order to be stored as my addresses, but they're not. The addresses under my account are empty.

https://github.com/woocommerce/woocommerce/pull/50934/commits/16e06cb3bd7c41a1ce1d064159a1ff582c4608fc

> When I'm logged in as an admin and place an order, I still see the Create new account form.

https://github.com/woocommerce/woocommerce/pull/50934/commits/90a6b26ffd4afe1522135fbd5a80b1525fb18961

File: plugins/woocommerce/changelog/add-delayed-account-creation-block-50632 (hunk 2 above)
Comment (mikejolley) [reply plugins/woocommerce/changelog/add-delayed-account-creation-block-50632:1]: "added or deprecated functionality in a backwards compatible way" is minor. major is breaking changes.

File: plugins/woocommerce-blocks/assets/js/blocks/order-confirmation/create-account/block.json (hunk 1 above)
Comment (mikejolley) [reply plugins/woocommerce-blocks/assets/js/blocks/order-confirmation/create-account/block.json:27]: tbh I see this being replaced by actual styling controls but I went with this method for now to offer parity with checkout.

**Review by alexflorisca — APPROVED** [review]
This looks good now, thanks for making those changes 👍

**Pull request merged**